src/utils/utils.hpp

```diff
@@ -54,6 +54,17 @@ namespace dd
       return true;
     }
 
+    static bool unique(int64_t val, std::vector<int64_t> vec)
```

Comment:
Maybe this is more compact: 
```
auto it = std::unique( a.begin(), a.end() );
bool wasUnique = (it == a.end() );
```
though this is not exactly what you are doing here.

Reply:
It seems to me that std::unique removes duplicates, which is not what I wanted to do, for some reason I do not remember